sysdump/sysdump.go

```diff
@@ -899,21 +899,27 @@ func (c *Collector) submitBugtoolTasks(ctx context.Context, pods []*corev1.Pod,
 			if len(m) != 2 || len(m[1]) == 0 {
 				return fmt.Errorf("failed to collect 'cilium-bugtool' output for %q in namespace %q: output doesn't contain archive name: %s", p.Name, p.Namespace, outString)
 			}
-			// Grab the resulting archive's contents from the pod.
-			b, err := c.Client.ExecInPod(ctx, p.Namespace, p.Name, containerName, []string{"cat", m[1]})
+
+			// Gzip bugtool tar file to reduce the size of transferred file
+			o, e, err = c.Client.ExecInPodWithStderr(ctx, p.Namespace, p.Name, containerName, []string{"gzip", m[1]})
```

Comment:
Not blocking but might be worth considering as a follow-up: Could we also change `cilium-bugtool` to write a `.tar.gz` directly? This way we wouldn't have to run `gzip` as part of the sysdump for Cilium versions where the bugtool writes a compressed archive.

Reply:
I completely agree with the above, the reason that I am not doing it right now is mentioned here https://github.com/cilium/cilium-cli/issues/616#issuecomment-1000023489.

Will raise one small ticket to track this if it's not urgent.

Reply:
Thanks for the link, seems I missed that comment. Makes sense to raise a follow-up ticket for this 👍